tests/examples/test_complex_examples.py

```diff
@@ -0,0 +1,78 @@
+"""Tests some of the more "complex" examples that require mocking or a temporary directory for artifacts."""
+
+import os
+from types import ModuleType
+
+
+def mock_config(config: ModuleType, overrides: dict):
+    """Helper function for overriding desired items in a config object."""
+    for key, value in overrides.items():
+        # sanity check
+        assert (
+            getattr(config, key, None) is not None
+        ), f"key '{key}' not found in config"
+        setattr(config, key, value)
+    return config
+
+
+def test_robot_bodybrain_cmaes(mocker):
+    """Test robot_bodybrain_cmaes example can complete."""
+    import examples.robot_brain_cmaes.config as config
+
+    overrides = {
+        "NUM_GENERATIONS": 2,
+    }
+    # override default config (to reduce number of generations)
+    mocker.patch(
+        "examples.robot_brain_cmaes.main.get_config",
+        return_value=mock_config(config, overrides),
+    )
+    import examples.robot_brain_cmaes.main as main_script
+
+    main_script.main()
```

Comment:
Should this file also use `assert_command_succeeds`?

Reply:
In this case it's needed because the "complex" examples are called directly as a python function, rathern than as a subprocess (thus have no exit code to check).  So in this case, the function not crashing or throwing an error is indicative of success (otherwise pytest will mark the test as failed).